### PR TITLE
Makes use of phase <> subphase db relationship

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9732

## Testing
**URL to test:** Moped test instance or local

**Steps to test:**
Navigate to a project's Timeline tab. Create a few phases with subphases. Verify subphases appear in dropdown and save correctly.

E.g., select a "potential" phase - your subphase options should be:

- Study recommended
- Study in progress
- Recommended
- Environmental study
- Active development review

For a "planned" phase, your subphase options should be:

- Environmental study
- Study in progress
- Ready for design
-

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/queries/project.js [hunk 1]
```diff
@@ -219,14 +219,17 @@ export const TIMELINE_QUERY = gql`
       phase_id
       phase_name
       phase_order
-      subphases
+      moped_subphases(order_by: { subphase_order: asc }) {
+        subphase_name
+        subphase_id
+      }
```
Comment (johnclary) [inline moped-editor/src/queries/project.js:225]: this is the only change to the query - the rest is linting

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 2]
```diff
@@ -266,39 +250,31 @@ const ProjectTimeline = ({ refetch: refetchSummary }) => {
...
       // We have a usable array of sub-phase ids, generate lookup values,
-      lookupValues = generateSubphaseLookup(
-        allowedSubphaseIds,
-        props.data.moped_subphases
-      );
+      lookupValues = allowedSubphases.reduce((obj, subphase) => {
+        obj[subphase.subphase_id] = subphase.subphase_name;
+        return obj;
+      }, {});
     }
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:274]: these two blocks are the substantive changes - the rest is linting

File: moped-database/metadata/tables.yaml [hunk 3]
```diff
@@ -641,6 +641,14 @@
 - table:
     schema: public
     name: moped_phases
+  array_relationships:
+    - name: moped_subphases
+      using:
+        foreign_key_constraint_on:
+          column: related_phase_id
```
Comment (johnclary) [inline moped-database/metadata/tables.yaml:648]: `related_phase_id` already existed - it just didn't have an fk constraint.

**Comment by johnclary** [comment]
It's worth noting that Chia wrote up a separate issue to address similar problems with `moped_proj_phases`: https://github.com/cityofaustin/atd-data-tech/issues/9350.

**Review by frankhereford — APPROVED** [review]
This is 🚀 🚢 ! 

I've tested this locally and it works just as described. I've looked over the code as well, but there are many folks with a better eye for JS than me, but it does look good. 👍

**Review by chiaberry — APPROVED** [review]

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 4]
```diff
@@ -144,30 +144,14 @@ const ProjectTimeline = ({ refetch: refetchSummary }) => {
     {}
   );
 
-  /**
-   * Generates a filtered number of sub-phase lookup elements
-   * @param {Array} allowedSubphaseIds - An array of integers containing the allowed subphase ids
-   * @param {Object} moped_subphases - The data object containing all sub-phase data
-   */
-  const generateSubphaseLookup = (allowedSubphaseIds, moped_subphases) =>
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:152]: 🎉

**Review by mddilley — APPROVED** [review]
Tested this out locally and it works as expected. I like seeing the Hasura and GraphQL take over some of this code logic! 💯

**Pull request merged**